### ISSUE TITLE
Model Architecture

### ISSUE THREAD

**Issue by anDoer** [BoyuanJiang/FitDiT#4]
Thanks for your amazing work! If have a few questions regarding your architecture.

In the paper you state that the customized DiT blocks are used only in the second stage: Does the GarmentDiT use unmodified MM-DiT blocks as in the SD3 paper? If so, what do you provide as input to the text condition branch? 

Have you removed the text condition branch from the Customized DiT architecture (as indicated in the Fig.2)?

**Comment by BoyuanJiang** [comment]
Thanks for following our work. For GarmentDiT, we also removed the text condition and used the clip image feature of garment as the condition just like DenoisingDiT. We will open source the inference code as soon as possible so that you can better understand our model structure.

**Comment by sunny926sun** [comment]
Thanks for your work.For GarmentDiT， <pooled_projections> is used pooled_image_embed(b, 1280),  and <encoder_hidden_states>  is set to None,  is that right?

**Comment by Manel-Hik** [comment]
this is an amazing work looking forward to the model weights

**Comment by sunny926sun** [comment]
Thanks for your reply, another question is :

**Comment by BoyuanJiang** [comment]
> Thanks for your reply, another question is : When training GarmentDiT, don't use the class free guidance , is that right ? and training DenoisingDiT, class free guidance is only use for the <pooled_projections> ?

For training GarmentDiT, no class free guidance. For training DenoisingDiT, both pooled_projections and garment latents will be dropout randomly.

**Comment by cryptexis** [comment]
hi @BoyuanJiang  excellent work, thank you for that!

Quick question on the GarmentDiT. Since the only conditioning is from pooled layers of the CLIP, I would assume it will never be able to learn perfectly reconstruct garments. The question is when enough is enough then ? :)

Could you please post and example of ground truth and reconstructed from GarmentDiT? Could you also pick the example at random? I am exactly interested not in how well it can do and rather how bad it could be. Many thanks in advance!

**Comment by BoyuanJiang** [comment]
hi @cryptexis , thanks for your question. It is indeed impossible to achieve perfect reconstruction of the input garment by using the clip vision feature, especially when a relatively large timesteps of noise is applied to the input garment. The figure below shows the reconstruction ability of GarmentDiT when 10% noise, 30% noise, 60% noise, 90% noise and full noise (i.e. using random noise as initialization). However, it is worth noting that the pretrain of our GarmentDiT is to better extract garment features rather than perfectly reconstruct the input garment, and when using it, feature extraction will only be performed when timestep=0.
![garment](https://github.com/user-attachments/assets/57d0ae59-27d9-45ab-af41-ddd9bcb5e1ba)

**Comment by sunny926sun** [comment]
@BoyuanJiang The reconstruction garment you show looks very good.  But i generate it use the weight you provide , the reconstruction image looks very blur. 

garm_img : 768 * 1024   
latents = noise * 0.3 + vae_garm_image * 0.7
50 timesteps:[1000.0000,  993.1244,  986.0571,  978.7898, ...  170.2957,  120.5327,  66.9002,    8.9286]   
then get last 15 timesteps : [548.0456, 522.8087, ...,  66.9002, 8.9286] 

for i in range(timesteps):
          latent_model_input = latents    (no cfg)
          noise_pred = self.transformer_garm(hidden_states=latent_model_input,
                                        timestep=timestep,
                                        pooled_projections=cloth_image_enbeds,
                                        encoder_hidden_states=None,
                                        return_dict=False)
          latents = self.scheduler.step(noise_pred, t, latents, return_dict=False)[0]

but the output is very blur (i set num_timestep=1000 is similar),   anything i set wrong ?  Thank you very much .
![garm (2)](https://github.com/user-attachments/assets/91df890b-6b9b-4073-9f87-2802995511dd)

**Comment by cryptexis** [comment]
> hi [@cryptexis](https://github.com/cryptexis) , thanks for your question. It is indeed impossible to achieve perfect reconstruction of the input garment by using the clip vision feature, especially when a relatively large timesteps of noise is applied to the input garment. The figure below shows the reconstruction ability of GarmentDiT when 10% noise, 30% noise, 60% noise, 90% noise and full noise (i.e. using random noise as initialization). However, it is worth noting that the pretrain of our GarmentDiT is to better extract garment features rather than perfectly reconstruct the input garment, and when using it, feature extraction will only be performed when timestep=0. ![garment](https://github.com/user-attachments/assets/57d0ae59-27d9-45ab-af41-ddd9bcb5e1ba)

thank you @BoyuanJiang makes a lot of sense. Then a natural question comes up - when is it a good time to stop the training? if we are not interested in the reconstruction accuracy?

**Comment by lizi85** [comment]
@BoyuanJiang I have tried running inference with the GarmentDiT weights and the stable diffusion 3 img2img pipeline and, similar to @sunny926sun, I also get a very blurry looking set of outputs:

![Image](https://github.com/user-attachments/assets/03bf49aa-ae67-4f03-9cf9-d3f6d7e892a0)

Would it be possible to release the inference code for GarmentDiT as well?
